theatre-js/theatre
# Publish development packages to npm

# Description

Build the `@theatre` packages based on the latest commit of the PR and publish them to npm (as `insider` packages).

# Versioning of the dev packages
A CI job builds the packages and assigns them a version number based on the version (without tags like `-dev.3`) specified in the root `package.json` (or `packages/r3f/package.json` in the case of `r3f` since that has an independent release schedule) and the abbreviated hash of the latest commit. Here's an example:
| prop | value |
| --- | --- |
| package name | `@theatre/core` |
| root version | `0.4.8-dev.3` |
| latest commit hash | `a369b71` |
| dev package version | `0.4.8-insiders.a369b71` |
| install command | `yarn add @theatre/core@0.4.8-insiders.a369b71` | 

# Testing this PR

1. Clone this repo: https://github.com/fulopkovacs/theatre-r3f-create-react-app
```
git clone git@github.com:fulopkovacs/theatre-r3f-create-react-app.git
```
2. Run the following commands in the repo (the second command takes quite a while):
```
yarn && yarn start
```
3. Visit `http://localhost:3000` in the browser
_Optional_:  You can verify that you're using the dev packages by checking the version of the `@theatre` packages with `yarn why <package-name>`, like:
```
$ yarn why @theatre/core
yarn why v1.22.10
[1/4] Why do we have the module "@theatre/core"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "@theatre/core@0.4.8-insiders.a369b71"
info Has been hoisted to "@theatre/core"
info This module exists because it's specified in "dependencies".
info Disk size without dependencies: "656KB"
info Disk size with unique dependencies: "1.13MB"
info Disk size with transitive dependencies: "3.73MB"
info Number of shared dependencies: 2
Done in 0.42s.
```

# CodeSandbox Example
Here's a CodeSandbox that uses the `insider` versions of the `@theatre packages`:
https://codesandbox.io/s/theater-js-r3f-using-insider-packages-2nvre0

# TODO after this PR is merged
Close https://github.com/theatre-js/theatre/pull/143.
